service/matching/task_queue_partition_manager.go

```diff
@@ -1075,3 +1085,31 @@ func (pm *taskQueuePartitionManagerImpl) getPerTypeUserData() (*persistencespb.T
 	perType := userData.GetData().GetPerType()[int32(pm.Partition().TaskType())]
 	return perType, userDataChanged, nil
 }
+
+// makePollerScalingDecision makes a decision on whether to scale pollers up or down based on the current state of the
+// task queue and the task about to be returned. Does not modify inputs.
+func (pm *taskQueuePartitionManagerImpl) makePollerScalingDecision(
```

Comment:
Should we make scaling decisions for forwarded polls? I don't see a clear reason to say yes or no yet...

Reply:
I'm not sure. This is why I had that open question about keeping track of the overall wait time including forwarding, since that would affect that I think.

Reply:
Good point. Though I'm not sure we actually need to pass the start time through forwarding: either a poll is forwarded immediately or not at all, so the wait time on the parent should be only slightly less than the wait time on the child